src/components/tabController/TabBar.tsx

```diff
@@ -128,364 +128,289 @@ interface Props extends TabControllerBarProps, BaseComponentInjectedProps, Forwa
+        uppercase,
+        iconColor,
+        selectedIconColor,
+        activeBackgroundColor,
+        ...child.props,
+        ...context,
+        index,
+        state: itemStates[index],
+        onLayout: onItemLayout
+      });
+    });
 
-  renderTabBarItems() {
-    const {itemStates} = this.context;
-    const {
-      labelColor,
-      selectedLabelColor,
-      labelStyle,
-      selectedLabelStyle,
-      uppercase,
-      iconColor,
-      selectedIconColor,
-      activeBackgroundColor
-    } = this.props;
-
-    if (_.isEmpty(itemStates)) {
-      return;
-    }
+  const renderTabBarItems = _.isEmpty(itemStates) ? null : items ? _renderTabBarItems : _renderTabBarItemsFromChildren;
 
-    if (this.items) {
-      return _.map(this.items, (item, index) => {
-        return (
-          <TabBarItem
-            labelColor={labelColor}
-            selectedLabelColor={selectedLabelColor}
-            labelStyle={labelStyle}
-            selectedLabelStyle={selectedLabelStyle}
-            uppercase={uppercase}
-            iconColor={iconColor}
-            selectedIconColor={selectedIconColor}
-            activeBackgroundColor={activeBackgroundColor}
-            key={item.label}
-            // width={this._itemsWidths[index]}
-            {...item}
-            {...this.context}
-            index={index}
-            state={itemStates[index]}
-            onLayout={this.onItemLayout}
-          />
-        );
-      });
-    } else {
-      // TODO: Remove once props.children is deprecated
-      if (this.tabBarItems) {
-        return this.tabBarItems;
-      }
+  const _indicatorWidth = new Value(0); // TODO: typescript?
+  const _indicatorOffset = new Value(0); // TODO: typescript?
 
-      if (!this.children) {
-        return;
-      }
+  const _indicatorTransitionStyle = {
+    // StyleProp<ViewStyle> TODO:
+    width: _indicatorWidth,
+    left: _indicatorOffset
+  };
 
-      this.tabBarItems = React.Children.map(this.children, (child: Partial<ChildProps>, index: number) => {
-        // @ts-ignore TODO: typescript - not sure if this can be easily solved
-        return React.cloneElement(child, {
-          labelColor,
-          selectedLabelColor,
-          labelStyle,
-          selectedLabelStyle,
-          uppercase,
-          iconColor,
-          selectedIconColor,
-          activeBackgroundColor,
-          ...child.props,
-          ...this.context,
-          index,
-          state: itemStates[index],
-          onLayout: this.onItemLayout
-        });
-      });
-      return this.tabBarItems;
-    }
-  }
+  const renderSelectedIndicator =
+    itemsWidths && itemsWidths.length > 0 ? (
+      <Reanimated.View style={[styles.selectedIndicator, indicatorStyle, _indicatorTransitionStyle]}/>
+    ) : undefined;
 
-  renderCodeBlock = _.memoize(() => {
-    const {currentPage, targetPage} = this.context;
-    const {itemsWidths, itemsOffsets} = this.state;
+  const renderCodeBlock = _.memoize(() => {
```

Comment:
_.memoize is a lodash memo util.. 
Not sure how it works in a react function component.
Maybe we would use useMemo here?

Reply:
I removed it and did not see unnecessary calls or performance problem, WDYT?

Reply:
ok, cool.. 
Let's keep it like that.